tutorial/Execution.md

```diff
@@ -0,0 +1,412 @@
+# A Random Walk Down Executions or: You Could Have Invented Executions or: Learn You An Execution For Greater Good
+
+## What is a Scalding Execution?
+
+A Scalding job (lowercase 'j') is the distributed completion of a DAG of Hadoop jobs that read, transform, and write data, usually to the Hadoop Distributed File System. A Scalding `Execution` is conceptually a plan to run zero or more Scalding jobs. The Scalding library has another paradigm, the Scalding `Job` class, that enables users to plan a single Scalding job in the constructor of a class. `Execution` is a newer, more composable, and more functional approach to running Scalding jobs.
+
+## Why would I want to use an Execution?
+
+The Scalding `Job` class uses a more object oriented programming approach to planning Scalding jobs, which often makes it more difficult to reason about and extend functionality. In addition, running multiple Scalding `Job` instances is poorly supported.
+
+Scalding `Execution`, on the other hand, is inspired by functional programming and composes easily. Here are a few things that are much easier with Scalding `Execution`:
+
+* Calling out to a service asynchronously to read or write some state before or after a Scalding job runs
+* Running multiple Scalding jobs or not running any Scalding jobs at all
+* Unit testing
+* Using the output of one Scalding job to decide how to plan future jobs
+
+These situations come up frequently when integrating Scalding jobs into an analytics workflow.
+
+We may want to avoid using Scalding `Execution` because it is more challenging to understand for newcomers to applied category theory (e.g. functors, applicative functors, and monads). This tutorial does not assume any knowledge of these concepts, but familiarity with them will make these explanations much easier to understand. In this respect, Scalding `Job` is simpler because it requires understanding less paradigms.
```

Comment:
So on the one may want to avoid Executions point, should we say that if you don't have a usecase that warrants it (e.g. calling a service before / after or iterative jobs etc) you could stick to using a job? Or do we want to nudge users towards Executions?